src/ants/wrapper.py

```diff
@@ -0,0 +1,352 @@
+                 modality='T1w',
+                 prob_threshold=0.5,
+                 num_threads=1,
+                 verbose=False):
+        """Initialize ANTs segmentation.
+        
+        Parameters
+        ----------
+        bids_dir : str or Path
+            Path to BIDS directory
+        output_dir : str or Path
+            Path to output directory
+        temp_dir : str or Path
+            Path to temporary directory
+        priors : list
+            List of paths to prior probability maps
+        modality : str
+            Modality to process (default: T1w)
+        prob_threshold : float
+            Probability threshold for binary mask creation
+        num_threads : int
+            Number of threads to use
+        verbose : bool
+            Whether to print detailed logs
+        """
+        self.bids_dir = Path(bids_dir)
+        self.output_dir = Path(output_dir)
+        self.temp_dir = Path(temp_dir) if temp_dir else self.output_dir / 'tmp'
+        self.priors = priors
+        self.modality = modality
+        self.prob_threshold = prob_threshold
+        self.num_threads = num_threads
+        self.verbose = verbose
+        
+        # Set up logging
+        self.logger = logging.getLogger('ants_bidsapp.segmentation')
+        
+        # Create necessary directories
+        if output_dir:
+            self.output_dir.mkdir(parents=True, exist_ok=True)
+        self.temp_dir.mkdir(parents=True, exist_ok=True)
+    
+    def load_image(self, image_path):
+        """
+        Load an image using ANTsPy.
+        
+        Parameters:
+        -----------
+        image_path : str
+            Path to the image file
+            
+        Returns:
+        --------
+        ants.ANTsImage
+            Loaded ANTs image
+        """
+        self.logger.info(f"Loading image: {image_path}")
+        try:
+            img = ants.image_read(image_path)
+            return img
+        except Exception as e:
+            self.logger.error(f"Failed to load image {image_path}: {str(e)}")
+            raise
+    
+    def preprocess_image(self, image):
+        """
+        Preprocess an ANTs image for segmentation.
+        
+        Parameters:
+        -----------
+        image : ants.ANTsImage
+            Input image to preprocess
+            
+        Returns:
+        --------
+        ants.ANTsImage
+            Preprocessed image
+        """
+        self.logger.info("Preprocessing image")
+        
+        # N4 bias field correction
+        self.logger.debug("Applying N4 bias field correction")
+        corrected = ants.n4_bias_field_correction(image)
+        
+        # Brain extraction
+        self.logger.debug("Performing brain extraction")
+        brain_mask = ants.get_mask(corrected)
+        brain_image = ants.mask_image(corrected, brain_mask)
+        
+        return brain_image
+    
+    def segment_image(self, image, priors=None, prior_weight=0.25, mrf=0.2, iterations=10):
```

Comment:
just an fyi that we want jointfusion segmentation, not just atropos segmentation.